src/Galley/FinchCompat/executor.jl

```diff
@@ -41,16 +41,100 @@ function (ctx::GalleyOptimizer)(prgm)
     julia_prgm
 end
 
-function Finch.set_options(ctx::GalleyOptimizer; estimator=DCStats)
+function Finch.set_options(ctx::GalleyOptimizer; estimator=DCStats, verbose=false)
     ctx.estimator=estimator
+    ctx.verbose=verbose
     return ctx
 end
 
+"""
+    get_stats_dict(ctx::GalleyOptimizer, prgm)
+
+Returns a dictionary mapping the location of input tensors in the program to their statistics objects.
+"""
+function get_stats_dict(ctx::GalleyOptimizer, prgm)
+    deferred_prgm = Finch.defer_tables(:prgm, prgm)
+    expr_stats_dict = Dict()
+    for node in PostOrderDFS(deferred_prgm)
+        if node.kind == table
+            expr_stats_dict[node.tns.ex] = ctx.estimator(node.tns.imm, [i.name for i in node.idxs])
+        end
+    end
+    return expr_stats_dict
+end
+
+"""
+    GalleyExecutor(ctx::GalleyOptimizer, tag=:global, verbose=false)
+
+Executes a logic program by compiling it with the given compiler `ctx`. Compiled
+codes are cached for each program structure. If the 'tag' argument is ':global', it maintains a set of plans 
+for inputs with different sparsity structures. In this case, it first checks the cache for a plan that
+was compiled for similar inputs and only compiles if it doesn't find one. If the `tag` argument is anything else,
+it will only compile once for that tag and will skip this search process.
+"""
+@kwdef struct GalleyExecutor
```

Comment:
I'd like the GalleyExecutor to have a configurable statistics similarity threshold, and to store the caches for each threshold and statistics type separately.

Also, I'd like to choose different executors for different compilation strategies (rather than using a sentinel tag value). Perhaps we can use the current executor for the "use first input strategy", and the GalleyExecutor for the "similar inputs" strategy. Then the GalleyExecutor wouldn't need a tag.